GliaWidgets/Lib/Coordinator/UIViewControllerCoordinator.swift

```diff
@@ -0,0 +1,75 @@
+import UIKit
+
+class UIViewControllerCoordinator: Coordinator {
+    private let id = UUID()
+    private var children = [UUID: Any]()
+
+    func start() -> UIViewController {
+        fatalError("start() must be implemented by subclass")
+    }
+
+    /// This method allows us to start child coordinator flows.
+    /// Memory management is automatic, meaning, if viewController gets deallocated
+    /// coordinator gets freed from memory as well. This is useful in cases where
+    /// we don't know when UIViewController will be deallocated (UINavigationController back
+    /// button press, dismiss modal controller with swipe down gesture)
+    func coordinate<T: UIViewControllerCoordinator>(to coordinator: T) -> UIViewController {
+        children[coordinator.id] = coordinator
+
+        let viewController = coordinator.start()
+
+        viewController.onDeinit = { [weak self] in
```

Comment:
Let's move deallocation handling to separate method. Since the main goal of this approach is to detect when view controller has gone off screen, we will be able to iterate on it separately, to improve it if possible.